torch/_inductor/lowering.py

```diff
@@ -3594,7 +3594,15 @@ def mask(index):
             if high != 0:
                 mask.append(range_mask_high(idx, length))
         mask = functools.reduce(ops.and_, mask)
-        return ops.masked(mask, lambda: x_loader(index), fill_value)
+        return ops.masked(
+            mask,
+            lambda: x_loader(index),
+            fill_value,
+            isinstance(x, TensorBox)
+            and isinstance(x.data, ir.StorageBox)
+            and x.data.is_input_buffer()
+            and x.get_numel() != 0,
+        )
```

Comment:
I don't really like this as it's extremely specific.
I think it makes more sense to:
- Always set `other=value` when the load is done in a masked region (hopefully this should be perf neutral... but would be good to confirm)
- See if we can remove the `where` when emitting `masked` by looking at the body of the masked region (for starters, when it's just composed of one load).